web/srwr_process.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+$path = './srwr_data/';
+$printrow = [];
+$companies = [];
+
+// Delete any previously downloaded files
+deleteFiles();
+
+// Fetch the zip file from the opendata
+fetchCSVFile();
+
+// Unzip the file to use the csv file
+$csv_file = unzipCSVFile();
+
+// Parse the data from the csv file
+while ($data = fgetcsv($csv_file)) {
+      collateRows($data);
+}
+
+// Attach the company names to the correct projects
+addCompanies();
+
+// Insert into the streetmanager database
+insertRows();
+
+function deleteFiles() {
+	global $path;
+	$files = glob($path . '*');
+	foreach($files as $file) {
+  		if(is_file($file)) {
+    		unlink($file);
+  		}
+	}
+}
+
+function fetchCSVFile() {
+	global $path;
+	$url = 'https://downloads.srwr.scot/export/api/v1/daily';
+
+	$ch = curl_init();
+    curl_setopt($ch, CURLOPT_URL, $url);
+    curl_setopt($ch, CURLOPT_FOLLOWLOCATION, true);
+    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+    $response = curl_exec($ch);
+    $httpcode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
+    if ($httpcode != 200) {
+		die("Error: Fetching url for zip file failed");
+    } else {
+		$data = json_decode($response, true);
+		$url = $data['url'];
+		curl_setopt($ch, CURLOPT_URL, $url);
+		$response = curl_exec($ch);
+		if ($httpcode != 200) {
+			die("Error: Fetching zip file failed");
+    	} else {
+			file_put_contents($path . 'daily.zip', $response);
+		}
+    }
+    curl_close($ch);
+    $ch = null;
+}
+
+function unzipCSVFile() {
+	global $path;
+
+	$zip = new ZipArchive();
+	$zipfiles = glob($path . '*.zip');
+	if (sizeof($zipfiles) != 1) {
+		die("Error: Expecting one zip file");
+	} else {
+		$zip->open($zipfiles[0]);
+		$zip->extractTo($path);
+		$csvfiles = glob($path . '*.csv');
+		if (sizeof($csvfiles) != 1) {
+			die("Error: Expecting one csv file");
+		} else {
+			$csv_file = fopen($csvfiles[0], 'r');
+			return $csv_file;
+		}
+	}
+}
+
+function collateRows($row) {
+	 global $printrow;
+	 global $companies;
+
+	 if ($row[1] == '006') {
+		$printrow[$row[2]]['project_ref'] = $row[3];
```

Comment:
If `$row[2]` is presumably the row ID, i'd have an `$id = $row[2]` first to make that clear